cmake/OpenCVCompilerOptimizations.cmake

```diff
@@ -392,17 +388,23 @@ elseif(PPC64LE)
 elseif(RISCV)
   option(RISCV_RVV_SCALABLE "Use scalable RVV API on RISC-V" ON)
 
-  ocv_update(CPU_RVV_TEST_FILE "${OpenCV_SOURCE_DIR}/cmake/checks/cpu_rvv.cpp")
   ocv_update(CPU_KNOWN_OPTIMIZATIONS "RVV")
-  ocv_update(CPU_RVV_FLAGS_ON "-march=rv64gcv")
+  ocv_update(CPU_RVV_TEST_FILE "${OpenCV_SOURCE_DIR}/cmake/checks/cpu_rvv.cpp")
+  ocv_update(CPU_RVV_FLAGS_ON "")
   if(RISCV_RVV_SCALABLE)
     set(CPU_RVV_FLAGS_ON "${CPU_RVV_FLAGS_ON} -DCV_RVV_SCALABLE")
   endif()
-  ocv_update(CPU_RVV_FLAGS_CONFLICT "-march=[^ ]*")
 
   set(CPU_DISPATCH "" CACHE STRING "${HELP_CPU_DISPATCH}")
   set(CPU_BASELINE "DETECT" CACHE STRING "${HELP_CPU_BASELINE}")
 
+  if(NOT CPU_BASELINE STREQUAL "DETECT")
+    message(FATAL_ERROR "RISC-V only supports CPU_BASELINE=DETECT, use CMAKE_CXX_FLAGS or cmake toolchain file options to change platform")
```

Comment:
I propose to revert changes in this files and use:
- https://github.com/opencv/opencv/pull/25930

Reply:
The main idea here is to avoid modifying _arch_ compiler flags at all. If I understand correctly, changes in #25930 will still try to append custom flags and run compiler checks.

For example, with that patch, if we compile OpenCV with `-DCMAKE_CXX_FLAGS="-mcpu=cortex-a72" -DCPU_BASELINE=NEON_FP16` options (`cortex-a72` is [armv8-a CPU](https://gcc.gnu.org/git/?p=gcc.git;a=blob;f=gcc/config/aarch64/aarch64-cores.def;h=e58bc0f27de3d60d39c02d2be2aa15570bd5db4d;hb=refs/heads/master)), then our scripts will silently append `-march=armv8.2-a+fp16` and make this build incompatible with explicitly requested CPU.

This behavior can cause compatibility, performance issues and compilation errors.

Reply:
> avoid modifying arch compiler flags at all

For dispatched optimization we still need to modify compiler flags anyway.

> make this build incompatible with explicitly requested CPU

If compiler doesn't fail on `try_compile()` in that case then it could be workaronded through `-DCPU_NEON_FP16_FLAGS_ON=` CMake override.

Reply:
Besides of DETECT, empty or other values of CPU_BASELINE are used for **limiting** of used features in OpenCV without touching compiler flags.

Reply:
> For dispatched optimization we still need to modify compiler flags anyway.

Yes, perhaps users will be required to manually provide flags for dispatched files via `CPU_XXX_FLAGS_ON` variables. Examples with our cmake-toolchain file from this PR:
* Build with RVV as a baseline: ` -DENABLE_RVV=ON`
* Build without RVV in a baseline, but with dispatched RVV files: `-DCPU_DISPATCH=RVV -DCPU_RVV_FLAGS_ON=-march=rv64gcv`

> empty or other values of CPU_BASELINE are used for limiting of used features

Doesn't `CPU_BASELINE_DISABLE` do this? I tried to use it with this PR and enabled RVV compiler flags: `-DENABLE_RVV=ON -DCPU_BASELINE_DISABLE=RVV`, but I'm not sure if correct macros are being set in cv_cpu_config.h:
```.h
// OpenCV CPU baseline features

#define CV_CPU_COMPILE_RVV 1
#define CV_CPU_BASELINE_COMPILE_RVV 1

#define CV_CPU_BASELINE_FEATURES 0 \
    , CV_CPU_RVV \


// OpenCV supported CPU dispatched features



#define CV_CPU_DISPATCH_FEATURES 0 \
```

Reply:
The goal is to avoid modifying compiler flags unnecessarily, as this can lead to incompatibilities and performance issues, especially when specific CPU architectures are requested. Users should have the option to manually set flags for dispatched files via CPU_XXX_FLAGS_ON. In the case of RVV, -DCPU_DISPATCH=RVV -DCPU_RVV_FLAGS_ON=-march=rv64gcv can handle dispatched optimizations without affecting the baseline configuration.